src/data/markdown/docs/02 javascript api/07 k6-experimental/03 websockets.md

```diff
@@ -5,22 +5,42 @@ excerpt: "k6 websockets experimental API"
+
+It's not fully implemented, but we're working on it. If you have any feedback, please [open an issue](https://github.com/grafana/xk6-websockets/) in the extension repository. Our long-term goal is to replace [`k6/ws`](/javascript-api/k6-ws/) module with this one.
+
+<!-- vale off -->
+| Class/Method | Description |
+| ------------ | ----------- |
+| [Params](/javascript-api/k6-experimental/websockets/params/)  | Used for setting various WebSocket connection parameters such as headers, cookie jar, compression, etc. |
+| [WebSocket(url, protocols, params)](/javascript-api/k6-experimental/websockets/websocket) | Constructs a new WebSocket connection. |
+| [WebSocket.close()](/javascript-api/k6-experimental/websockets/websocket/websocket-close) | Close the WebSocket connection. |
+| [WebSocket.ping()](/javascript-api/k6-experimental/websockets/websocket/websocket-ping) | Send a ping. |
+| [WebSocket.send(data)](/javascript-api/k6-experimental/websockets/websocket/websocket-send) | Send string data. |
+| [WebSocket.addEventListener(event, callback)](/javascript-api/k6-experimental/websockets/websocket/websocket-addeventlistener) | Add an event listener on the connection for specific event. |
+| [WebSocket.onmessage(callback)](/javascript-api/k6-experimental/websockets/websocket/websocket-onmessage) | Set up an event listener on the connection for `message` event |
+| [WebSocket.onerror(callback)](/javascript-api/k6-experimental/websockets/websocket/websocket-onerror) | Set up an event listener on the connection for `error` event |
+| [WebSocket.onopen(callback)](/javascript-api/k6-experimental/websockets/websocket/websocket-onopen) | Set up an event listener on the connection for `open` connection event |
+| [WebSocket.onclose(callback)](/javascript-api/k6-experimental/websockets/websocket/websocket-onclose) | Set up an event listener on the connection for `close` connection event |
+| [WebSocket.onping(callback)](/javascript-api/k6-experimental/websockets/websocket/websocket-onping) | Set up an event listener on the connection for `ping` event |
+| [WebSocket.onpong(callback)](/javascript-api/k6-experimental/websockets/websocket/websocket-onpong) | Set up an event listener on the connection for `pong` event |
+<!-- vale on -->
+
+A WebSocket instance also has the following properties:
+
+| Class/Property | Description |
+| -------------- | ----------- |
+| WebSocket.readyState | The current state of the connection. Could be one of [the four states](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/readyState). |
+| WebSocket.url | The URL of the connection as resolved by the constructor. |
+| WebSocket.bufferedAmount | The number of bytes of data that have been queued using calls to `send()` but not yet transmitted to the network. |
+| WebSocket.binaryType | The [`binaryType`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/binaryType) is by default `ArrayBuffer`. Setting it panics, as k6 does not support Blob. |
```

Comment:
"Panic" is a Go term.

```suggestion
| WebSocket.binaryType | The [`binaryType`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/binaryType) is by default `"arraybuffer"`. Setting it throws an exception, as k6 does not support the Blob type. |
```

Reply:
:+1: 

Good one! I'll take part about panic removal, but for the `ArrayBuffer` it seems like we currently return it that way https://github.com/grafana/xk6-websockets/blob/main/websockets/websockets.go#L184 even if the specification defines it as `arraybuffer`. 

I took it as a local TODO since an I have a local branch that I'm going to push soon with some chore fixes.
```suggestion
| WebSocket.binaryType | The [`binaryType`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/binaryType) is by default `"ArrayBuffer"`. Setting it throws an exception, as k6 does not support the Blob type. |
```